### PR TITLE
ADBDEV-6410: Migrate gpbackup tests for GPDB 7 in Ubuntu-based container

### PR DESCRIPTION
**Pull request by silent-observer**
Migrate gpbackup tests for GPDB 7 in Ubuntu-based container

Fix the docker test to support Ubuntu for GPDB 7 by adding a case testing the
current OS. Also modify the README and add comment in Dockerfile to reflect
migration to Ubuntu.

### CODE REVIEW

**Comment by RekGRpth** [comment]
> current OS. Also modify the README and ad comment in Dockerfile to reflect

current OS. Also modify the README and **add** comment in Dockerfile to reflect

File: arenadata/run_gpbackup_tests.bash [hunk 1]
```diff
@@ -2,9 +2,23 @@
...
+		yum install -y glibc-locale-source || true
+		;;
+	ubuntu*)
+		apt-get install -y locales
+		;;
+	*)
+		echo "Unknown OS: '$OS'"
+		exit 1
+		;;
+esac
```
Comment (RekGRpth) [inline arenadata/run_gpbackup_tests.bash:21]:
```suggestion

yum install -y glibc-locale-source || true
case "$TEST_OS" in
	rocky*)
		yum install -y glibc-locale-source
		;;
esac
```

**Comment by RekGRpth** [comment]
Why didn't the test on the 7 pass? It doesn't look like a flaky test.
```
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[ERROR]:-Error loading data into table public.tht: COPY tht, line 3: "9,39": ERROR: value of distribution key doesn't belong to segment with ID 1, it belongs to segment with ID 0  (seg1 172.17.0.2:7003 pid=39959) (SQLSTATE 23000)
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[ERROR]:-Encountered 1 error(s) during table data restore; see log file /home/gpadmin/gpAdminLogs/gprestore_20241119.log for a list of table errors.
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[INFO]:-Data restore completed with failures
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[INFO]:-Restoring post-data metadata
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[DEBUG]:-Post-data objects restored:  30% (1/3)
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[DEBUG]:-Post-data objects restored:  60% (2/3)
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[DEBUG]:-Post-data objects restored:  100% (3/3)
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[INFO]:-Post-data metadata restore complete
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[INFO]:-Found neither /usr/local/greenplum-db-devel/bin/gp_email_contacts.yaml nor /home/gpadmin/gp_email_contacts.yaml
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[INFO]:-Email containing gprestore report /tmp/temp1747023088/demoDataDir-1/backups/20241119/20241119121159/gprestore_20241119121159_20241119121201_report will not be sent
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[INFO]:-Beginning cleanup
20241119:12:12:01 gprestore:gpadmin:runner-dat56y4g-project-1187-concurrent-0:039935-[INFO]:-Cleanup complete
------------------------------
• [FAILED] [3.000 seconds]
backup and restore end to end tests Properly handles enum type as distribution or partition key [It] Restores table data partitioned using GPDB7 partition syntax
/home/gpadmin/go/src/github.com/greenplum-db/gpbackup/end_to_end/end_to_end_suite_test.go:2230
  [FAILED] exit status 1
  In [It] at: /home/gpadmin/go/src/github.com/greenplum-db/gpbackup/end_to_end/end_to_end_suite_test.go:310 @ 11/19/24 12:12:01.782
------------------------------
••••••••••••••••••••••••••••S••••••S••••••••••••••••••••••S•••••••••••••••••••••••••••••••••••••••••••••••••••S••••••••••••••••••••••S•••••••••••••••••••••••••••••••••••••S••••••••••
Summarizing 1 Failure:
  [FAIL] backup and restore end to end tests Properly handles enum type as distribution or partition key [It] Restores table data partitioned using GPDB7 partition syntax
  /home/gpadmin/go/src/github.com/greenplum-db/gpbackup/end_to_end/end_to_end_suite_test.go:310
Ran 205 of 211 Specs in 1781.556 seconds
FAIL! -- 204 Passed | 1 Failed | 0 Pending | 6 Skipped
--- FAIL: TestEndToEnd (1781.56s)
FAIL
Ginkgo ran 1 suite in 29m44.258540099s
Test Suite Failed
```

**Comment by silent-observer** [comment]
Yes, seems to be a [known](https://tracker.yandex.ru/ADBDEV-6055) flaky test. Restarting the job.

**Pull request merged**